File: 2015/ruby/test/test_day01.rb
require 'test/unit'
require_relative '../lib/day01'

class TestDay01 < Test::Unit::TestCase
  def test_part1
    assert_equal 0, Day01.part1('(())')
    assert_equal 3, Day01.part1('(()(()(')
    assert_equal -3, Day01.part1(')())())')
  end

  def test_part2
    assert_equal 1, Day01.part2(')')
    assert_equal 5, Day01.part2('()())')
  end
end

File: 2016/python/tests/test_day01.py
import unittest

from collections import namedtuple

from days.day01 import part1, part2


class TestDay01(unittest.TestCase):
    def test_part1(self):
        TestCase = namedtuple("TestCase", ["input", "expected"])
        cases = [
            TestCase("R2, L3", 5),
            TestCase("R2, R2, R2", 2),
            TestCase("R5, L5, R5, R3", 12),
        ]

        for tc in cases:
            self.assertEqual(part1(tc.input), tc.expected)

    def test_part2(self):
        input = "R8, R4, R4, R8"
        expected = 4
        self.assertEqual(part2(input), expected)
